service/api/interfaces.go

```diff
@@ -5,6 +5,12 @@ import (
 	"github.com/xdblab/xdb-apis/goapi/xdbapi"
 )
 
+type Server interface {
+	// Start will start running on the background
+	Start() error
+	Stop(ctx context.Context) error
+}
+
 // Service is the interface of API service, which decoupled from REST server framework like Gin
 // So that users can choose to use other REST frameworks to serve requests
 type Service interface {
```

Comment:
Can we reduce the the abstraction of interface here given these 2 reasons:
1. we only have one service, and not likely to suppor one more service with the same interface.
2. the goland ide could not directly jump to the implementation with this interface layer. Removing it will make it easier to search the codes in IDE.

Reply:
> the goland ide could not directly jump to the implementation with this interface layer. Removing it will make it easier to search the codes in IDE.

It actually works for me. Lets sync up on this later. 

Unlike some lang like Java, adding interface to struct is quite standard in Golang. For example, in Java you can just write unit tests to mock a class. In Golang, you cannot mock any struct. You have to add an interface and mock the interface instead. 

Interfaces are needed not for unit tests, but also for cleaner code. It's important to have an interface if the code is being shared to other package to use. For example, the Server interface is being used in `bootstrap`. This `Service` interface is going to be used for Indeed internal implementation of Server -- because internally we use other built in REST framework which comes with metrics/logging.